salt/state.py

```diff
@@ -2002,12 +2005,17 @@ def reconcile_procs(self, running):
                                'changes': {}}
                     try:
                         with salt.utils.fopen(ret_cache, 'rb') as fp_:
-                            ret = msgpack.loads(fp_.read())
+                            ret = msgpack.loads(fp_.read(), encoding='utf-8')
```

Comment:
Could we use `__salt_system_encoding__` here instead of forcing the encoding to `utf-8`?